File: cinema-backend/src/main/java/org/example/cinemabackend/movie/application/dto/response/MovieResponse.java
package org.example.cinemabackend.movie.application.dto.response;

import lombok.Builder;
import org.example.cinemabackend.cinema.application.dto.response.ProjectionTechnologyResponse;
import org.example.cinemabackend.movie.core.domain.AgeRestriction;
import org.example.cinemabackend.movie.core.domain.Genre;

import java.time.LocalDate;
import java.util.Set;

@Builder
public record MovieResponse(
        String title,
        String originalTitle,
        Integer duration,
        LocalDate releaseDate,
        String description,
        String poster,
        ProductionDetailsResponse productionDetails,
        SubtitleAndSoundOptionsResponse subtitleAndSoundOptions,
        AgeRestriction ageRestriction,
        VideoFileResponse trailer,
        Set<Genre> genres,
        Set<ProjectionTechnologyResponse> projectionTechnologies
) {
}

File: cinema-customer-frontend/src/app/repertory/dtos/response/movie.response.ts
import {AgeRestrictionResponse} from "./age-restriction.response";
import {ProductionDetailsResponse} from "./production-details.response";
import {MovieVariantResponse} from "./movie-variant.response";

export class MovieResponse {
  constructor(
    readonly title: string,
    readonly originalTitle: string,
    readonly duration: number,
    readonly releaseDate: Date,
    readonly productionDetails: ProductionDetailsResponse,
    readonly description: string,
    readonly ageRestriction: AgeRestrictionResponse,
    readonly poster: string,
    readonly trailer: string,
    readonly genres: string[],
    readonly movieVariants: MovieVariantResponse[]
  ) {
  }
}
